### PR TITLE
Modify telemetry test "cluster type detection" function

### PR DESCRIPTION
**Pull request by MaxBab**
**Please provide a description of this PR:**
This is a followup PR for https://github.com/istio/istio/pull/54348

When "Image-regirector" deployment is created on Openshift cluster and kubeconfig file provided via KUBECONFIG environment variable and the "~/.kube/config" file does not exist, the "IsKindCluster" function fails to allocate the kubeconfig file from the env var.

Modify the function to check the "KUBECONFIG" env var and check the kubeconfig file path from it.

### CODE REVIEW

**Comment by MaxBab** [comment]
Hello @howardjohn @lei-tang 

I would appreciate you review in this PR.
This is a follow up PR for the https://github.com/istio/istio/pull/54348 with a small fix.

Thank you.

File: tests/integration/telemetry/api/setup_test.go [hunk 1]
```diff
@@ -161,8 +162,16 @@ proxyMetadata:
 // By looking into a context name. Expects "kind-" prefix.
 // That is required by some tests for specific actions on "Kind".
 func IsKindCluster() (bool, error) {
-	config, err := clientcmd.LoadFromFile(clientcmd.RecommendedHomeFile)
+	kubeconfig := os.Getenv("KUBECONFIG")
```
Comment (howardjohn) [inline tests/integration/telemetry/api/setup_test.go:165]: Hmm, thinking about this more -- this is looking at the current context pointed at, but that isn't always how the test is configured. For multiclsuter, for instance, ` --istio.test.kube.topology=opology-config.json` is used to point to a set of clusters that are unrelated to the current context.

The most reliable way would be to somehow get this from `(resource.Context).Clusters()`. I don't know how to get from there to the info we need, though.

Comment (MaxBab) [reply tests/integration/telemetry/api/setup_test.go:165]: From what I could see, the multicluster test has been executed as part of this gate.
And I could see that it loads the cluster topology as - `/home/prow/go/src/istio.io/istio/prow/config/topology/multicluster.json`. And it looks like it is not affected by this change.

Maybe I could modify the logic of the function, so in case of an error to load the kubeconfig file, it will return `true` as a fallback to be indicated as kind cluster.

Comment (howardjohn) [reply tests/integration/telemetry/api/setup_test.go:165]: I am not concerned with the CI we have -- it clearly works, likely because it happens to point to the kind cluster it created. But in theory you can (and I have) run things locally like this in a manner it wouldn't work.

I am 100% fine with not handling every obscure edge case :slightly_smiling_face: just wanted to make it known that there are edge cases

Comment (MaxBab) [reply tests/integration/telemetry/api/setup_test.go:165]: You're right.

Unfortunately, as per now, I could not think of any proper solution.
But from the other side, without this PR, CI prow Openshift job will fail during execution. :(

**Review by howardjohn — APPROVED** [review]

**Review by lei-tang — APPROVED** [review]

File: tests/integration/telemetry/api/setup_test.go (hunk 1 above)
Comment (MaxBab) [reply tests/integration/telemetry/api/setup_test.go:165]: Where could I find the example structure  of `--istio.test.kube.topology=opology-config.json` file?
I will try to see how it work with this patch and what could be done with it.
